crates/sui-types/src/messages.rs

```diff
@@ -1932,12 +1932,14 @@ pub struct ConsensusTransaction {
 pub enum ConsensusTransactionKey {
     Certificate(TransactionDigest),
     CheckpointSignature(AuthorityName, CheckpointSequenceNumber),
+    EndOfPublish(AuthorityName),
```

Comment:
Since you would be checking the author name anyway when a message is received, why including the name here? Could just use the message author name?

Reply:
In theory yes, but adding name here makes code easier, for example I can construct proper ConsensusTransactionKey in ConsensusTransaction::key method.

I think in the long term we will refactor the code a bit and have some wrapper type around ConsensusTransaction, something like `ReceivedConsensusTransaction {from, transaction}`, and then we could re-organize code and remove this field

Reply:
cool perhaps include this information as comment there in case anyone has the same question